pkg/gadgets/interface.go

```diff
@@ -140,9 +135,12 @@ func (f *BaseFactory) Delete(name string) {
 		factory.DeleteTrace(name, trace)
 	}
 	delete(f.traces, name)
-	return
 }
 
 func (f *BaseFactory) Operations() map[string]TraceOperation {
 	return map[string]TraceOperation{}
 }
+
+func (f *BaseFactory) OutputModesSupported() map[string]struct{} {
+	return map[string]struct{}{}
+}
```

Comment:
It's more a question for @alban. I'm wondering why we have to implement this method in the base interface if this is gonna be redefined in all gadgets. 

According to my understanding (after hitting some compilation errors), it's because the line `factory, ok := TraceFactory(f).(TraceFactoryWithDeleteTrace)` above. (`BaseFactory` needs to implement `TraceFactory`). 

Is there any particular reason to use this approach instead of using `factory, ok := reflect.TypeOf(f).(TraceFactoryWithDeleteTrace)` and removing the definitions for `Operations` and `OutputModesSupported` in `BaseFactory`?

Reply:
I didn't know about reflect.TypeOf(). If that really works, then we can do that. I would want to test that the correct Delete() is really called in that case.

Reply:
I can experiment this later on. It's fine for me as it's now.